main-settings/src/main/scala/sbt/Structure.scala

```diff
@@ -17,7 +17,18 @@ import sbt.Def.{ Initialize, KeyedInitialize, ScopedKey, Setting, setting }
 import std.TaskExtra.{ task => mktask, _ }
 
 /** An abstraction on top of Settings for build configuration and task definition. */
-sealed trait Scoped { def scope: Scope; val key: AttributeKey[_] }
+sealed trait Scoped {
+  def scope: Scope
+  val key: AttributeKey[_]
+
+  override def equals(that: Any) =
```

Comment:
Wouldn't this create cooperative equality on Scoped subtypes?
If you want equals, I think they should be implemented at the class, not sealed trait.

Reply:
yes! thank you! I'll change it and add a test that enforces it.